### ISSUE TITLE
Stamper Duping / incorrectly calculating recipes

### ISSUE THREAD

**Issue by Ligands** [DaedalusGame/Soot#44]
Affects either / any stamp. ~~I've got a crafttweaker script to melt Constantan ingots/plates into 144 mb of molten constantan, as well as stamp one ingot or plate from 144 mb with the respective stamp.~~ -
 nothing to do with crafttweaker- it affects all materials. The only requirement is that the stamper must have a stamp 'equipped' while it's receiving fluid- the bug does not occur if the fluid is pumped in while no stamp is set.

Not quite sure what's going on with the stamper- I have a fluid gauge attached, and after it stamps out the first ingot, it displays '24/1000', suggesting that it's only consumed 120 mb, even though it should have used 144mb- and it then continues to produce another ingot with that remaining 24 mb... Furthermore, if I put 3 ingots in the melter, I actually get 4 ingots back out the stamper, so it's more like the bug's only occurring on the 'final' stamp rather than an actual 'dupe'...

**Versions:**
Forge: 14.23.3.2675
Soot: 0.15

**Comment by DaedalusGame** [comment]
jesus what 
i will investigate this soonish

**Comment by Ligands** [comment]
Hang on. Just noticed this has nothing to do with crafttweaker! I can dupe any ingot by simply re-melting it. Investigating some more...

**Comment by Ligands** [comment]
Okay, I've isolated the mods to eliminate any chance of external incompatibilities. Been looking into it & it seems to get weirder: it only affects stampers that have already have a stamp while receiving a fluid to stamp out. The dupe does not occur if I load up the internal buffer of the stamp base, and only then attach a bar stamp after I've finished pumping the fluids over.

It affects any materials / stamps, and occurs regardless of whether there's a bin under the stamp base- as long as the stamper has a stamp equipped to it before the fluid is inserted.
